python/MooseDocs/extensions/MooseImageFile.py

```diff
@@ -0,0 +1,103 @@
+import re
+import os
+from markdown.util import etree
+import logging
+log = logging.getLogger(__name__)
+
+import MooseDocs
+from markdown.inlinepatterns import Pattern
+from MooseCommonExtension import MooseCommonExtension
+import utils
+
+class MooseImageFile(MooseCommonExtension, Pattern):
+    """
+    Markdown extension for handling images.
+    """
+
+    # Find !image /path/to/file attribute=setting
+    RE = r'!image\s+(.*?)\s+(.*)'
```

Comment:
I think we need to mandate this to be on a new line by itself, otherwise we will probably have problems.